pulsar/consumer_impl.go

```diff
@@ -317,6 +319,7 @@ func (c *consumer) Close() {
 		wg.Wait()
 		close(c.closeCh)
 	})
+	c.client.handlers.Del(c)
```

Comment:
Is there a reason not to call this in the closeOnce?

Reply:
the initial thought was `delete` operation will be no-op if consumer/producer not in handlers, so i didnt put `Del` call in the `closeOnce`. But your idea is better, which will prevent non necessary locks&unlocks.